Documentation using gh-pages (#57)
open-energy-transition/form-energy-storage: documentation/02-tutorials.rst

##########################################
Tutorials
##########################################

This is a brief tutorial on how to use the Form Energy Storage scenarios.
Unlike the original PyPSA-Eur repository, this version comes pre-configured with scenarios that you can test run.

Build the tutorial model
=========================

First, open the ``config.form.yaml`` file using a code editor of your choice, and navigate to the ``run`` section:

.. literalinclude:: ../config/config.form.yaml
   :language: yaml
   :start-at: run:
   :end-before: # docs

> as you can see, ``baseline-mds`` is the name of the scenario that is chosen. 

Reviewer: We may want to add a list of scenarios here (or a reference to a list)

The content of ``baseline-mds`` is defined in ``scenarios.form.yaml``.

.. literalinclude:: ../config/scenarios.form.yaml
   :language: yaml
   :start-at: baseline-mds:
   :end-before: baseline-nomds:

To create your own scenarios, it's recommended to define the configurations in ``scenarios.form.yaml`` by either modifying an existing scenario or duplicating and renaming a scenario.

.. note::
    If you want to create a model with lower computational requirements, you can adjust the ``clustering: temporal: resolution_sector`` setting here:

    .. literalinclude:: ../config/config.form.yaml
       :language: yaml
       :start-at: clustering:
       :end-before: # docs

>     The rule of thumb is:

Reviewer: There is something missing here

    final_adjustment_myopic                                 1
    prepare_network                                         1
    prepare_sector_network                                  1
    retrieve_cost_data                                      1
    retrieve_electricity_demand                             1
    retrieve_gas_infrastructure_data                        1
    retrieve_osm_prebuilt                                   1
    retrieve_ship_raster                                    1
    simplify_network                                        1
    solve_sector_network_myopic                             1
    solve_sector_networks                                   1
    time_aggregation                                        1
    total                                                  68

.. note::
    If an error occurs due to missing data, the first solution is to check the ``enable:`` setting in ``config.form.yaml``.

    .. literalinclude:: ../config/config.form.yaml
       :language: yaml
       :start-at: enable:
       :end-before: # docs

    * ``retrieve_databundle``: set this to true if a file in ``data/`` is missing

>     * ``retrieve_cost_data``: set this to true if ``resources/.. ../cost_2035.csv`` is missing

Reviewer: Improve path reference here as well

    * ``retrieve_cutout``: set this to true if ``cutouts/europe-2013-sarah3-era5.nc`` is missing

    For cutouts from years other than 2013, 2018, and 2023, you can download them from xx.

To create the model, run the snakemake command:

.. code:: console

   snakemake solve_sector_networks --configfile config/config.form.yaml

This triggers a workflow of multiple preceding jobs that depend on each rule's inputs and outputs.

Analyze the model
=========================

There are several ways to analyze the results of the Snakemake run:

* By using the CSV files and graphs located in ``results/../csvs/`` and ``results/../graphs/``

> * By using a pre-configured Jupyter Notebook in ``notebooks/`` or any Jupyter Notebook of your choice.

Reviewer: Are we intending to deliver this ?

Author: Well, we could still share everything other than the network itself right?

##########################################
Tutorials
##########################################

> This is a brief tutorial on how to use the Form Energy Storage scenarios.

Reviewer 2: ```suggestion
This is a brief tutorial on how to recreate the scenarios of the study **The Role of Energy Storage in Germany**.
```

Unlike the original PyPSA-Eur repository, this version comes pre-configured with scenarios that you can test run.

Build the tutorial model
=========================

> First, open the ``config.form.yaml`` file using a code editor of your choice, and navigate to the ``run`` section:

Reviewer 2: ```suggestion
First, open the base configuration file for this study, called ``config.form.yaml`, using a code editor of your choice, and navigate to the ``run`` section:
```

.. literalinclude:: ../config/config.form.yaml
   :language: yaml
   :start-at: run:
   :end-before: # docs

as you can see, ``baseline-mds`` is the name of the scenario that is chosen. 

> The content of ``baseline-mds`` is defined in ``scenarios.form.yaml``.

Reviewer 2: ```suggestion
The content of ``baseline-mds`` as well as a comprehensive list of all studied scenarios is defined in ``scenarios.form.yaml`` as described in [Scenarios Configuration](https://open-energy-transition.github.io/form-energy-storage/12-scenarios.html).
```

    retrieve_cost_data                                      1
    retrieve_electricity_demand                             1
    retrieve_gas_infrastructure_data                        1
    retrieve_osm_prebuilt                                   1
    retrieve_ship_raster                                    1
    simplify_network                                        1
    solve_sector_network_myopic                             1
    solve_sector_networks                                   1
    time_aggregation                                        1
    total                                                  68

.. note::
    If an error occurs due to missing data, the first solution is to check the ``enable:`` setting in ``config.form.yaml``.

    .. literalinclude:: ../config/config.form.yaml
       :language: yaml
       :start-after: #enable
       :end-before: # docs

    * ``retrieve_databundle``: set this to true if a file in ``data/`` is missing
    * ``retrieve_cost_data``: set this to true if ``resources/{run}/cost_2035.csv`` is missing
    * ``retrieve_cutout``: set this to true if ``cutouts/europe-2013-sarah3-era5.nc`` is missing

>     For cutouts from years other than 2013, 2018, and 2023, you can download them from xx.

Reviewer 2: ```suggestion
    For cutouts of the years 1996, 2010 and 2012, you can download them from xx and place into the cutouts folder of the repository.
```

> To create the model, run the snakemake command:

Reviewer 2: ```suggestion
To create and solve the model, run the snakemake command:
```

.. code:: console

   snakemake solve_sector_networks --configfile config/config.form.yaml

This triggers a workflow of multiple preceding jobs that depend on each rule's inputs and outputs.

> 

Reviewer 2: ```suggestion
To create and solve the model and automatically generate all KPI plots of the study, run the snakemake command:

.. code:: console
   snakemake all --configfile config/config.form.yaml

```
